File: PORC-Front/src/models/pub_account_info/account_cust/AccountCustomisation.ts
import { scheduleToRecv, type Schedule } from "../../schedule/Schedule";
import type { Shiftstone } from "../../shiftstone/ShiftstoneModel";
import type { RadarChart } from "./radar_chart/RadarChart";

export interface AccountCustomisation {
    region: string | null,
    bp: number,
    radar_chart: RadarChart | null,
    shiftstones: [Shiftstone | null, Shiftstone | null],
    badges: number[],
    unlocked_badges: number[],
    banner: string | null
}

File: src/liberary/account_lib/account/account_customisation.rs
use serde::{Serialize, Deserialize};

use crate::liberary::account_lib::account::user_data::{radar_chart::RadarChart, shiftstone::Shiftstone};

#[derive(Clone, Serialize, Deserialize, Debug)]
pub struct AccountCustomisation {
    pub region: Option<String>,
    pub bp: u32,
    pub radar_chart: Option<RadarChart>,
    pub shiftstones: (Option<Shiftstone>, Option<Shiftstone>),
    pub badges: Vec<u64>,
    pub unlocked_badges: Vec<u64>,
    pub banner: Option<String>
}